src/ObjectStateMutations.js

```diff
@@ -135,7 +135,8 @@ export function estimateAttributes(
           const fields = attr.split('.');
           const first = fields[0];
           const last = fields[fields.length - 1];
-          data[first] = { ...serverData[first] };
+          // data[first] = { ...serverData[first] };
+          data[first] = (serverData[first] === undefined) ? {} : serverData[first];
```

Comment:
This line is buggy. It should be doing a deep-copy. Note that it is buggy even on master, except the master does a deep-copy on the first level, thanks to the spread operator (see [the Mozilla docs on the spread operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax) ) and this patch does a complete shallow copy.
The shallow copy causes estimateAttributes() to actually change serverData as a side effect, though only for nested fields. The result is that a mere call to get() changes the object attributes, for nested fields with pending increments.
I will be creating a new issue for this.